packages/celotool/src/e2e-tests/validator_handshake_tests.ts

```diff
@@ -0,0 +1,263 @@
+import { eqAddress } from '@celo/utils/lib/address'
+import { assert } from 'chai'
+import _ from 'lodash'
+import sleep from 'sleep-promise'
+import Web3 from 'web3'
+import { Admin } from 'web3-eth-admin'
+import { GethRunConfig } from '../lib/interfaces/geth-run-config'
+import { GethInstanceConfig } from '../lib/interfaces/geth-instance-config'
+import { configureInstances, getContext, jsonRpc, killInstance, restartInstance } from './utils'
+
+const VALIDATORS = 3
+const PROXIED_VALIDATORS = 1
+const EPOCH = 20
+
+const TMP_PATH = '/tmp/e2e'
+
+describe('Validator handshake tests', () => {
+  const gethConfig: GethRunConfig = {
+    networkId: 1101,
+    network: 'local',
+    runPath: TMP_PATH,
+    migrate: false,
+    instances: getInstances(),
+    genesisConfig: {
+      epoch: EPOCH,
+    },
+  }
+  const context: any = getContext(gethConfig)
+
+  before(async function(this: any) {
+    this.timeout(0)
+    await context.hooks.before()
+  })
+
+  after(context.hooks.after)
+
+  describe('Validator handshake', () => {
+    before(async function() {
+      // reset the instances for each test
+      gethConfig.instances = getInstances()
+      configureInstances(gethConfig)
```

Comment:
Is this something we should be doing for other tests? If not could you expand on the comment as to why this type of reset is needed?

Reply:
Added a comment, it's because the tests are modifying the geth config (like changing proxy config, or the max peers)